test/linalg.jl

```diff
@@ -40,6 +40,8 @@ using StaticArrays, Base.Test
 
     @testset "diagm()" begin
         @test @inferred(diagm(SVector(1,2))) === @SMatrix [1 0; 0 2]
+        @test @inferred(diagm(SVector(1,2,3), Val{2})) == diagm([1,2,3], 2)
```

Comment:
A minor thing - might be worth adding a type assert here `::SMatrix` on the left hand side outside the `@inferred` to assert that these come out as static arrays.